src/test/java/org/javarosa/xpath/test/XPathEvalTest.java

```diff
@@ -523,6 +523,10 @@ public void doTests() {
         testEval("id-compress(0, 'CD','','ABCDE',1)", null, ec, new XPathException());
         testEval("id-compress(0, 'CD','CD','ABCDE',1)", null, ec, new XPathException());
 
+        testEval("checksum('verhoeff','41310785898')", null, null, "4");
```

Comment:
I took some sample Aadhar ids and passed the first 11 digits to the checksum function while the last digit as the expected outcome. So for this case Aadhar Id I used would be  '413107858984".